controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -295,6 +308,66 @@ public void getCurrentSegments(StreamInfo request, StreamObserver<SegmentRanges>
                 responseObserver);
     }
 
+    private Supplier<String> createDelegationTokenSupplier(StreamInfo request) {
+        return () -> {
+            StreamAuthParams authParams = new StreamAuthParams(request.getScope(), request.getStream(),
+                    this.isInternalWritesWithReadPermEnabled);
+
+            // StreamResource will take on values containing `_RG` (ex: "prn:://scope:myScope/stream:_RGmyApp")
```

Comment:
The `streamResource` represents the resource in a "stream", even for reader groups (`prn:://scope:myScope/stream:_RGmyApp`) and watermarks that we otherwise treat as resources in their own right (like `prn:://scope:myScope/reader-group:myApp`). 

The `streamResource` (like `prn:://scope:myScope/stream:_RGmyApp`) is what is used in the claims in the delegation token and Segment Store processes it upon appends/reads. However, the Controller uses `prn:://scope:myScope/reader-group:myApp`  for any authorization. I kept it that way so that the Segment Store doesn't need to perform a lot of translation in the read/write path to account for the differentiation.